task-impls/src/network.rs

```diff
@@ -72,11 +69,44 @@ impl<
         COMMCHANNEL: CommunicationChannel<TYPES, Message<TYPES, I>, PROPOSAL, VOTE, MEMBERSHIP>,
     > NetworkTaskState<TYPES, I, PROPOSAL, VOTE, MEMBERSHIP, COMMCHANNEL>
 {
+    /// Handle the given message.
+    pub async fn handle_message(&mut self, message: Message<TYPES, I>) {
+        let event = match message.kind {
+            MessageKind::Consensus(consensus_message) => match consensus_message.0 {
+                Either::Left(general_message) => match general_message {
+                    GeneralConsensusMessage::Proposal(proposal) => {
+                        SequencingHotShotEvent::QuorumProposalRecv(
+                            proposal.clone(),
+                            proposal.signature,
+                        )
+                    }
+                    GeneralConsensusMessage::Vote(vote) => {
+                        SequencingHotShotEvent::QuorumVoteRecv(vote.clone(), vote.signature())
+                    }
+                    _ => panic!("Got unexpected message type in network task!"),
+                },
+                Either::Right(committee_message) => match committee_message {
+                    CommitteeConsensusMessage::DAProposal(proposal) => {
+                        SequencingHotShotEvent::DAProposalRecv(proposal.clone(), proposal.signature)
+                    }
+                    CommitteeConsensusMessage::DAVote(vote) => {
+                        SequencingHotShotEvent::DAVoteRecv(vote.clone(), vote.signature.1)
+                    }
+                },
+            },
+            MessageKind::Data(_) => {
+                panic!("Got unexpected message type in network task!");
```

Comment:
It might be better to log an error here and return `None`. Otherwise the adversary could send a data message and cause the entire networking task to fail.

Reply:
I changed it to `warn!` but didn't make it return `None`, since this function doesn't have a return value. Let me know if you think it's better to return something!

Reply:
Oops! I confused this with the message handler type. Yes, this makes sense!